etcdctl/ctlv3/command/make_mirror_command.go

```diff
@@ -54,6 +57,7 @@ func NewMakeMirrorCommand() *cobra.Command {
 	c.Flags().StringVar(&mmcert, "dest-cert", "", "Identify secure client using this TLS certificate file for the destination cluster")
 	c.Flags().StringVar(&mmkey, "dest-key", "", "Identify secure client using this TLS key file")
 	c.Flags().StringVar(&mmcacert, "dest-cacert", "", "Verify certificates of TLS enabled secure servers using this CA bundle")
+	c.Flags().BoolVar(&mmusediscoverysrv, "dest-use-discovery-srv", false, "Interpret destination as SRV discovery domain")
 	// TODO: secure by default when etcd enables secure gRPC by default.
```

Comment:
Maybe just `dest-discovery-srv` to alight with other srv flags.

Reply:
If it's `--dest-discovery-srv` then what does a user provide as `arg[0]`?

So the UI of this subcommand is different to the standard approach.

The usage would make a lot more sense if the endpoints were specified as options and the prefixes are arguments, i.e.:

`usage: etcdctl make-mirror [options] source-prefix [dest-prefix]`

That way `--{,dest-}discovery-srv` and `--{,dest-}endpoints` could be mixed and matched along with all other options.

Reply:
You know I have never seen this feature to be honest :).  But seems like we could do.

```
etcdctl  make-mirror --endpoints $src --dest-discovery-srv example.com
```
or in theory

```
etcdctl make-mirror --discovery-srv source.com --dest-discovery-srv dest.com
```

if `--dest-discovery-srv` has value then we don't take/need arg[0] as the value from the key is what we are using. Instead of adding a bool then working against arg[0]. What do you think?

Reply:
https://github.com/etcd-io/etcd/blob/329be66e8b3f9e2e6af83c123ff89297e49ebd15/etcdctl/ctlv3/command/make_mirror_command.go#L64-L66

Reply:
@jsok thanks for the notes. Can we drop the naked args altogether and just use flags?

`--endpoints` `--dest-endpoints`? I will try to take a deeper look at this soon but maybe we can take this chance to make this feature not such a special snowflake. What do you think?